### ISSUE TITLE
Publishing a Python app fails

### ISSUE THREAD

**Issue by asavaritayal** [Azure/azure-functions-core-tools#1514]
Basic hello-world function fails to publish with the following error: 

![image](https://user-images.githubusercontent.com/8435148/63398162-8751e700-c381-11e9-8c3c-5aea1962a50e.png)

I think the client-side build is trying to reference `pip` while my machine only supports `pip3`.

**Comment by ankitkumarr** [comment]
Ah! I think you are right. `packapp` starts a subprocess `pip`. We could probably try `pip3` first and if it's not there, fallback to `pip`. Because in case, both are available, `pip` may be for Python2.

Ideally, we should do `python3.6 -m pip ...`. Core-tools knows the right binary to use. So, either core-tools can pass that info to `packapp`, or `packapp` can use [sys.executable](https://docs.python.org/3/library/sys.html#sys.executable) and do something like `f'{sys.executable} -m pip ...`.

I think we should try the `sys.executable` approach.

Thanks @asavaritayal.

**Comment by ankitkumarr** [comment]
If anyone is blocked on this, they should be able to instead use --
1.  `func azure functionapp publish <appname> --build remote`, or
2. `func azure functionapp publish <appname> --build-native-deps` (needs Docker)

And if they really really want to use packapp (client side build without Docker), they can manually change references of `pip` to `pip3` in our script, for now. From the above screenshot, go to `/usr/local/Cellar/azure-functions-core-tools/2.7.1575/tools/python/packapp/__main__.py`, and change all references of `'pip'` to `'pip3'`.

\cc @jeffhollan @anirudhgarg

**Comment by yorickdewid** [comment]
Why is this still an open issue? Pip3 is the default package manager for Python3 in linux environment. If anything detect `pip3` or fallback to `pip` otherwise

**Comment by kshyju** [comment]
Closing this issue due to inactivity for over 4 years. If you are still experiencing an issue with the latest stable version of the core tools, please open a new issue with relevant details. Thank you!
